### ISSUE TITLE
Node 'output/softmax' does not exist in model 'file:///android_asset/OptimizedGraph.pb'

### ISSUE THREAD

**Issue by bibhu107** [tensorflow/tensorflow#19526]
I trained my model then frozen the graph using the .ckpt file and .pbtxt file  
**Code to freeze:**
 

    freeze_graph.freeze_graph('./tensorflowModel.pbtxt', "",False,'./tfmodel.ckpt', "output/predictions", "save/restore_all",  "save/Const:0",'frozen.pb', True,"")

Then to **optimize for inference** followed the  below code:

    inputGraph = tf.GraphDef()
    with tf.gfile.Open("frozen.pb", "rb") as f:
        data2read = f.read()
        inputGraph.ParseFromString(data2read)
    
    outputGraph = optimize_for_inference_lib.optimize_for_inference(
                    inputGraph,
                    ["inputTensor"], # an array of the input node(s)
                    ["output/predictions"], # an array of output nodes
                    tf.int32.as_datatype_enum)
    
            # Save the optimized graph
    
    f = tf.gfile.FastGFile("OptimizedGraph.pb", "w")
    f.write(outputGraph.SerializeToString())  

 

The output file **"OptimizedGraph.pb",** When I try to use it in the android studio it produces the error:

     Node 'output/softmax' does not exist in model 'file:///android_asset/OptimizedGraph.pb'

I checked my .pbtxt file which I used to freeze my graph and there was no node **output/softmax**!

What should I do?

**Comment by achalshah20** [comment]
Can you post your android code where you load the model and do inference.

**Comment by sonu107** [comment]
@achalshah20  There is another optimized.pb file with me and that works perfectly fine, but when I am following the above-mentioned steps to create another new .pb file, It shows error.

**Comment by sonu107** [comment]
```
public class TensorflowIntegrationExample {
    private Classifier classifier;
    private final String TAG = "TestModel";
    private final String INPUT_NAME = "inputTensor";
    private final String DROPOUT_NAME = "dropout_keep_prob";
    private final String[] OUTPUT_NAME = {"output/softmax"};
    private final String MODEL_FILE = "file:///android_asset/OptimizedGraph.pb";
    private final String LABEL_FILE = "file:///android_asset/labels.txt";
    private AssetManager assetManager;
```

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by bibhu107** [comment]
The issue is solved closing it! Thank you @achalshah20

**Comment by towcar** [comment]
What was the solution?

**Comment by bibhu107** [comment]
@towcar  Well, I had not declared my softmax layer. If you are facing the same check if the node "output/softmax " exists or not.
